diff --git a/pdns/ws-api.cc b/pdns/ws-api.cc
index cea1abd78..5dc0c9a66 100644
--- a/pdns/ws-api.cc
+++ b/pdns/ws-api.cc
@@ -365,7 +365,7 @@ string apiZoneNameToId(const DNSName& dname) {
 }
 
 void apiCheckNameAllowedCharacters(const string& name) {
-  if (name.find_first_not_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890_/.-") != std::string::npos)
+  if (name.find_first_not_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890_/.-+!@#$%^&*()[]{}=\\|<>?;:~") != std::string::npos)
     throw ApiException("Name '"+name+"' contains unsupported characters");
 }
 
